src/y_array.rs

```diff
@@ -388,35 +387,32 @@ impl YArray {
         match &mut self.0 {
             SharedType::Integrated(array) => {
                 let doc = array.doc.clone();
-                let origin = Origin::from(uuid_v4().to_string());
-                array
-                    .inner
-                    .observe_deep_with(origin.clone(), move |txn, events| {
-                        Python::with_gil(|py| {
-                            let events = events_into_py(txn, events, doc.clone());
-                            if let Err(err) = f.call1(py, (events,)) {
-                                err.restore(py)
-                            }
-                        })
-                    });
-                Ok(ObservationId(origin))
+                let subscription = array.inner.observe_deep(move |txn, events| {
+                    Python::with_gil(|py| {
+                        let events = events_into_py(txn, events, doc.clone());
+                        if let Err(err) = f.call1(py, (events,)) {
+                            err.restore(py)
+                        }
+                    })
+                });
+                Ok(ObservationId(subscription))
             }
             SharedType::Prelim(_) => Err(PreliminaryObservationException::default_message()),
         }
     }
 
     /// Cancels the callback of an observer using the `observation_d` returned from the `observe` method.
-    pub fn unobserve(&mut self, observation_d: ObservationId) -> PyResult<bool> {
+    pub fn unobserve(&mut self, observation_d: ObservationId) -> PyResult<()> {
```

Comment:
Wouldn't be better to implement `close` method on `ObservationId`? This way you could potentially use it in `with` statements. Not a single thing in `unobserve` method requires `self` to be a param either.